app/src/main/java/com/kinandcarta/create/proxytoggle/feature/manager/viewmodel/ProxyManagerViewModel.kt

```diff
@@ -14,7 +15,8 @@ import com.kinandcarta.create.proxytoggle.settings.AppSettings
 class ProxyManagerViewModel @ViewModelInject constructor(
     private val deviceSettingsManager: DeviceSettingsManager,
     private val proxyValidator: ProxyValidator,
-    private val appSettings: AppSettings
+    private val appSettings: AppSettings,
+    private val themeSwitcher: ThemeSwitcher
```

Comment:
so this class is acting as a repository and a reference/controller of the ui?

Reply:
Well, this class collects the click actions from the view (and acts accordingly) and also exposes the data that the UI needs to show. I don't think I'd call it a Repository unless you have in mind another definition for what a Repository means?

Reply:
AppSettings is a data source - Repositories are classes or components that encapsulate the logic required to access data sources... ThemeSwitcher seems to directly control the view - instead of having the view model provide it with actions and data/state, which is what a view model is responsible of doing. That's why this class looks like a view controller and repository to me

Reply:
Not sure I follow your point.
Are you suggesting the ViewModel should expose a state based on the ThemeSwitcher's value, and then the view should change the theme itself?

Also, I am using AppSettings interface to expose the last used proxy. If I create a Repository, it will only be for a setter/getter of one value - and it will never grow. 
Given the size of this app (and the fact that it won't grow more than this) and given that the AppSettings implementation has its own setters/getters from the SharedPrefs, do you think it's necessary creating a Repository for just this?

Reply:
Not saying anything is necessary... I was just trying to understand the role of this class, as is it isn't a view model

Reply:
Why is it not a ViewModel? Can a ViewModel handle the user clicks, run some small logic like input validation, execute UseCases and/or access Repository data?

Reply:
Part of the definition of a view model in MVVM is "The viewmodel exposes methods, commands, and other points that help maintain the state of the view, manipulate the model as the result of actions on the view, and trigger events in the view itself". This class you have a directly calling a view controller, the ThemeSwitcher class you have. I don't know what ProxyValidator and DeviceSettingsManager are

Reply:
If you make ThemeSwitcher and interface then I would say this class is somewhat a Presenter in MVP

Reply:
Well, this class is triggering events in the View itself, isn't it? The ThemeSwitcher is part of the view and it's triggering the toggleTheme event.

DeviceSettingsManager is basically where the Proxy gets set and stored. You could call this a Repository if you want. The only "special" thing is that this DSM knows about the android framework because interacts directly with the device.

I have the feeling that we're going in circles because of some naming. I am happy to review/rename whatever you consider it's not well named. Or maybe you could open the discussion to the whole CoP and see what people think? 

I am not saying that you're wrong, but let's keep in mind this is a single screen application that only sets one value, using something you "shouldn't" use because is system-protected. Maybe it isn't the ideal project to be an architecture purist. But again, happy to beat it up, discuss and rename classes if that makes more sense!

Reply:
> If you make ThemeSwitcher an interface then I would say this class is somewhat a Presenter in MVP

Can definitely do that! 
Although, aren't the ViewModel in MVVM and the Presenter in MVP a similar piece of the architecture, in terms of responsibilities? They have obvious differences when it comes to implementation, but they are still the same thing for the View

Reply:
I didn't say you need to change something. You could even just write everything in one class if this is going to be so small. No need for clean architecture
This class isn't triggering events though, it is calling them directly, for example `themeSwitcher.toggleTheme()`. A view model to be a view model would be called by the view or receive some data from a data source and then emit a new state that will trigger the change in the view. Also looking at `ThemeSwitcher` it is also directly calling a data source

Reply:
There's lots of material on the web about what a view model is in MVVM, including official documentation. MVVM, MVP, MVC, MVI can be applied on any system with a view from Android, Web and iOS. They didn't originate from Android. This Stackoverflow post is one of the shorter resource shared at companies I have worked at to get a better understanding of MVVM https://stackoverflow.com/questions/5421874/basic-concepts-of-mvvm-what-should-a-viewmodel-do

Reply:
If you have some other official definition of a view model in MVVM specifically for Android, please feel free to pm me and share. I am happy to learn :)

Reply:
As I said, let's take this offline and we can raise a separate PR for addressing the architecture/naming convention :)